test/backend/DataSourceTestSuite.scala

```diff
@@ -59,10 +63,12 @@ class DataSourceTestSuite extends PlaySpec {
             defaultViewConfiguration = None,
             adminViewConfiguration = None,
             coordinateTransformations = None,
-            numChannels = Some(1),
             additionalAxes = Some(Seq(AdditionalAxis("time", bounds = Seq(0, 5), 3))),
-            attachments = Some(DatasetLayerAttachments(
-              meshes = Seq(LayerAttachment("meshfile", new URI("./meshes/meshfile"), LayerAttachmentDataformat.zarr3))))
+            attachments = Some(
+              DataLayerAttachments(
+                meshes = Seq(LayerAttachment("meshfile",
+                                             UPath.fromStringUnsafe("./meshes/meshfile"),
+                                             LayerAttachmentDataformat.zarr3))))
```

Comment:
could you please add an inequality check below?

Kinda alike
``` 
val notEqualDataSource = dataSourceCopyPastedFromAbove.copy(name="testLayerChanged")
assert(dataSource.hashCode() != notEqualDataSource.hashCode())
```
maybe?

Reply:
Good idea, will do!